### ISSUE TITLE
[Enhancement] Add more values to the NamedSize enum to more fully support accessibility on iOS

### ISSUE THREAD

**Issue by johnshardman** [xamarin/Xamarin.Forms#12842]
The NamedSize enum currently supports the following values in order to support Dynamic Type (Larger Text accessibility option) on iOS:

NamedSize.Body
NamedSize.Caption
NamedSize.Header
NamedSize.Subtitle
NamedSize.Title

However, on iOS, UIFont now supports all of the following:

PreferredHeadline
PreferredSubheadline
PreferredFootnote
PreferredCaption1
PreferredCaption2
PreferredTitle1
PreferredTitle2
PreferredTitle3
PreferredCallout
PreferredBody

NamedSize (and related renderers) should be updated to support the full set of Dynamic Type sizes, in order to provide a wider range of sizes that support accessibility.

**Comment by rachelkang** [comment]
Hi, @johnshardman - thanks for the proposal and for helping shed light on this issue! In investigating various options for making font sizes more accessible, particularly on iOS, we have decided to make all fonts [autoscale by default in .NET MAUI](https://github.com/dotnet/maui/pull/1774) and included a `FontAutoScalingEnabled` property that developers can use to disable the autoscaling as well. Unfortunately, this would be a breaking change we cannot proceed with in Xamarin.Forms at this time, but the implemention in .NET MAUI, the next iteration of Xamarin.Forms, ensures that font sizes will be much more accessible!
